File: src/api/services/AuthService.ts

import axios from 'axios';
import { User, UserRole } from '../models';
import { API_URL } from '../../config';
export interface LoginCredentials {
  email: string;
  password: string;
}

export interface SignupData extends LoginCredentials {
  name: string;
  phone?: string;
}

export const AuthService = {
  async login(credentials: LoginCredentials): Promise<{ user: User; token: string }> {
    try {
      const response = await axios.post(`${API_URL}/auth/login`, credentials);

      // Store token in localStorage
      if (response.data.token) {
        localStorage.setItem('auth_token', response.data.token);

        // Set default role if not provided
        if (!response.data.user.role) {
          response.data.user.role = UserRole.HOSPITAL_ADMIN;
        }
      }

      return response.data;
    } catch (error) {
      console.error('Login error:', error);
      throw error;
    }
  },

  async signup(userData: SignupData): Promise<{ user: User; token: string }> {
    try {
      const response = await axios.post(`${API_URL}/auth/signup`, userData);

      // Store token in localStorage
      if (response.data.token) {
        localStorage.setItem('auth_token', response.data.token);

        // Set default role if not provided
        if (!response.data.user.role) {
          response.data.user.role = UserRole.HOSPITAL_ADMIN;
        }
      }

      return response.data;
    } catch (error) {
      console.error('Signup error:', error);
      throw error;
    }
  },

  logout(): void {
    localStorage.removeItem('auth_token');
  },

  getToken(): string | null {
    return localStorage.getItem('auth_token');
  },

  isAuthenticated(): boolean {
    return !!this.getToken();
  }
};

File: myclinic-dispensary-app/lib/services/auth_service.dart
import '../config/api_config.dart';
import '../models/user.dart';
import 'api_service.dart';
import 'storage_service.dart';

class AuthService {
  final _api = ApiService();
  final _storage = StorageService();

  Future<User> login(String email, String password) async {
    final response = await _api.post(ApiConfig.loginAdmin, data: {
      'email': email,
      'password': password,
    });

    final data = response.data;
    final token = data['token'];
    final userData = data['user'];

    await _storage.saveToken(token);
    await _storage.saveUser(userData);

    return User.fromJson(userData);
  }

  Future<User?> getMe() async {
    try {
      final response = await _api.get(ApiConfig.getMe);
      final userData = response.data['user'] ?? response.data;
      await _storage.saveUser(userData);
      return User.fromJson(userData);
    } catch (_) {
      return null;
    }
  }

  Future<void> changePassword(
      String currentPassword, String newPassword) async {
    await _api.post(ApiConfig.changePassword, data: {
      'currentPassword': currentPassword,
      'newPassword': newPassword,
    });
  }

  Future<void> updateProfile(String name) async {
    await _api.put(ApiConfig.updateMe, data: {'name': name});
  }

  Future<void> logout() async {
    await _storage.clearAll();
  }

  Future<bool> isLoggedIn() async {
    final token = await _storage.getToken();
    return token != null;
  }
}
